### PR TITLE
*: visible default resource group

### LINKED ISSUE #40875: Make default resource group visible and changable
## Enhancement
For resource control feature https://github.com/pingcap/tidb/issues/38825 we need to initialize `default` resource group at cluster bootstrap time and make it visible from `information_schema.resource_groups` and support `alter resource group` command.

### PR DESCRIPTION
**Pull request by BornChanger**
### What problem does this PR solve?

Issue Number: close #40875

Problem Summary:

### What is changed and how it works?

Initialize the default resource group to resource manager, and let it readable and changeable  but not droppable. 

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
Simulate the upgrade from 6.6 to nightly to verify default resource group will be created.
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by BornChanger** [comment]
client-go pr is https://github.com/pingcap/tidb/issues/40875.  This pr needs to be refreshed when the client-go pr is merged.

File: ddl/resourcegroup/group.go [hunk 1]
```diff
@@ -22,6 +22,12 @@ import (
 // MaxGroupNameLength is max length of the name of a resource group
 const MaxGroupNameLength = 32
 
+// MinRUFillRate defines the lower boundary of RU fill rate
+const MinRUFillRate = 1
+
+// MaxRUFillRate defines the higher boundary of RU fill rate
+const MaxRUFillRate = 10000000
```
Comment (nolouch) [inline ddl/resourcegroup/group.go:29]: I think we need reliable reasons for this setting. :(

Comment (nolouch) [reply ddl/resourcegroup/group.go:29]: cc @easonn7

File: ddl/resource_group_test.go [hunk 2]
```diff
@@ -54,6 +56,10 @@ func TestResourceGroupBasic(t *testing.T) {
 
 	tk.MustExec("set global tidb_enable_resource_control = 'on'")
 
+	tk.MustExec("alter resource group `default` ru_per_sec=10000")
+	tk.MustQuery("select * from information_schema.resource_groups where name = 'default'").Check(testkit.Rows("default 10000 NO"))
+	tk.MustContainErrMsg("drop resource group `default`", "can't drop internal resource groups")
```
Comment (glorv) [inline ddl/resource_group_test.go:61]:
```suggestion
	tk.MustContainErrMsg("drop resource group `default`", "can't drop reserved resource group")
```

File: ddl/resourcegroup/group.go [hunk 3]
```diff
@@ -30,10 +36,14 @@ func NewGroupFromOptions(groupName string, options *model.ResourceGroupSettings)
 	if len(groupName) > MaxGroupNameLength {
 		return nil, ErrTooLongResourceGroupName
 	}
+
 	group := &rmpb.ResourceGroup{
 		Name: groupName,
 	}
 	if options.RURate > 0 {
+		if options.RURate > MaxRUFillRate || options.RURate < MinRUFillRate {
```
Comment (nolouch) [inline ddl/resourcegroup/group.go:44]: I think we don't need to limit the MaxRUFillRate and MinRUFillRate. cc @glorv

**Review by nolouch — APPROVED** [review]
rest lgtm

**Review by tiancaiamao — APPROVED** [review]

File: ddl/resourcegroup/group.go (hunk 3 above)
Comment (glorv) [reply ddl/resourcegroup/group.go:44]: +1

**Review by glorv — APPROVED** [review]

File: session/bootstrap.go [hunk 4]
```diff
@@ -2368,6 +2373,8 @@ func upgradeToVer135(s Session, ver int64) {
 
 	mustExecute(s, "INSERT HIGH_PRIORITY IGNORE INTO %n.%n VALUES (%?, %?);",
 		mysql.SystemDB, mysql.GlobalVariablesTable, variable.TiDBOptAdvancedJoinHint, false)
+
+	doReentrantDDL(s, CreateDefaultResourceGroup)
```
Comment (bb7133) [inline session/bootstrap.go:2377]: Why do we add `CreateDefaultResourceGroup` to `upgradeToVer135`? Usually, when we add a new upgrade code, it is recommended to add a new `upgradeToVer...`.

Comment (BornChanger) [reply session/bootstrap.go:2377]: It's a result of rebase. L2374 is introduced in coming 7.0, which shares the version 135.

File: session/bootstrap.go [hunk 5]
```diff
@@ -825,6 +828,8 @@ const (
 	version135 = 135
 	// version136 prepare the tables for the distributed task.
 	version136 = 136
+	// version137 introduces some reserved resource groups
+	version137 = 137
```
Comment (bb7133) [inline session/bootstrap.go:832]: Please bump `currentBootstrapVersion` to `version137`.

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**